### PR TITLE
Stop using deprecated rust action

### PR DESCRIPTION
**Pull request by emlowe**

### CODE REVIEW

**Review by richardkiss — APPROVED** [review]
Seems fine I suppose.

File: .github/workflows/build-test.yml [hunk 1]
```diff
@@ -194,10 +190,8 @@ jobs:
         with:
           fetch-depth: 1
       - name: Install rust
-        uses: actions-rs/toolchain@v1
+        uses: dtolnay/rust-toolchain@nightly
```
Comment (richardkiss) [inline .github/workflows/build-test.yml:193]: Do we want `stable` downloaded too? Not sure if it does that.

Comment (emlowe) [reply .github/workflows/build-test.yml:193]: I was assuming the previous action
```
uses: actions-rs/toolchain@v1
        with:
            toolchain: nightly
            override: true
            components: rustfmt, clippy
```
already only did the nightly version but perhaps it didn't work that way

Comment (richardkiss) [reply .github/workflows/build-test.yml:193]: You can definitely have both toolchains installed simultaneously. Might be worth understanding why nightly is requested for this step. That said, it probably is fine if there are no errors (and nightly builds are allowed to break stuff with no notice anyway).

**Review by Rigidity — APPROVED** [review]
lgtm

File: .github/workflows/build-test.yml (hunk 1 above)
Comment (arvidn) [reply .github/workflows/build-test.yml:193]: I believe some feature used to require the nightly build, but I would not be surprised those features have made it into stable by now

**Review by arvidn — APPROVED** [review]

**Pull request merged**